crates/mipsy_interactive/src/interactive/commands/mod.rs

```diff
@@ -54,20 +54,20 @@ impl From<ArgumentKind> for i64 {
     }
 }
 
-// TODO: another callback for __help__ label?
+// TODO: remove cmd callback params. find another way because currently its only use is for getting subcommands
 // TODO: remove once if-let chaining is in
 #[derive(Clone)]
 pub(crate) struct Argument {
     name: String,
-    sanitiser: fn(arg: &str, helper: &MyHelper) -> CommandResult<ArgumentKind>,
-    hints: fn(harg: &HintArgs, helper: &MyHelper) -> Vec<String>,
+    sanitiser: fn(cmd: &Command, arg: &str, helper: &MyHelper) -> CommandResult<ArgumentKind>,
+    hints: fn(cmd: &Command, harg: &HintArgs, helper: &MyHelper) -> Vec<String>,
```

Comment:
not really ideal to keep `cmd` param since its really only used for subcommand hinting. another way would be to make the callbacks be like a `&dyn Fn` instead to capture the command as a parameter but since both of them are called once per input i wasnt sure the performance impact of changing it